scripts/setup-k8s-node.sh.in

```diff
@@ -30,8 +30,18 @@ kubeadm reset --force
 # Get the discovery file
 aws s3 cp "s3://{{ClusterInfoBucket}}/cluster-info.yaml" /tmp/cluster-info.yaml
 
-kubeadm join \
-    --node-name="$(hostname -f 2>/dev/null || curl http://169.254.169.254/latest/meta-data/local-hostname)" \
-    --token="{{ClusterToken}}" \
-    --discovery-file="/tmp/cluster-info.yaml" \
-    "{{K8sMasterPrivateIp}}:6443"
+HOSTNAME="$(hostname -f 2>/dev/null || curl http://169.254.169.254/latest/meta-data/local-hostname)"
+
+cat >/tmp/kubeadm-node.yaml <<EOF
+---
+apiVersion: kubeadm.k8s.io/v1alpha2
+kind: NodeConfiguration
```

Comment:
Is NodeConfiguration available in v1alpha2?

Reply:
I got the parameters from here https://github.com/kubernetes/kubernetes/blob/release-1.11/cmd/kubeadm/app/apis/kubeadm/v1alpha2/types.go#L234

Reply:
It's also mentioned in the docs https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-join/ so I guess it is :)

Reply:
oh yeah, my fault, totally missed that